tools/image-builder/locals.tf

```diff
@@ -0,0 +1,341 @@
+locals {
+  # Defined values
+  name           = "hpc-packer"
+  prefix         = var.prefix
+  tags           = [local.prefix, local.name]
+  no_addr_prefix = true
+  # Derived values
+  vpc_id = var.vpc_name == null ? module.landing_zone.vpc_data[0].vpc_id : data.ibm_is_vpc.existing_vpc[0].id
+  # Resource group calculation
+  # If user defined then use existing else create new
+  create_resource_group = var.resource_group == "null" ? true : false
+  resource_groups = var.resource_group == "null" ? [
+    {
+      name   = "${local.prefix}-service-rg",
+      create = local.create_resource_group,
+      use_prefix : false
+    },
+    {
+      name   = "${local.prefix}-workload-rg",
+      create = local.create_resource_group,
+      use_prefix : false
+    }
+    ] : [
+    {
+      name   = var.resource_group,
+      create = local.create_resource_group
+    }
+  ]
+  # For the variables looking for resource group names only (transit_gateway, key_management, atracker)
+  resource_group = var.resource_group == "null" ? "${local.prefix}-service-rg" : var.resource_group
+  region         = join("-", slice(split("-", var.zones[0]), 0, 2))
+  zones          = ["zone-1", "zone-2", "zone-3"]
+  active_zones = [
+    for zone in var.zones :
+    format("zone-%d", substr(zone, -1, -2))
+  ]
+  packer_sg_variable_cidr_list = split(",", var.network_cidr)
+  address_prefixes = {
+    "zone-${element(split("-", var.zones[0]), 2)}" = [local.packer_sg_variable_cidr_list[0]]
+  }
+
+  # Subnet calculation
+  active_subnets = {
+    for zone in local.zones : zone => contains(local.active_zones, zone) ? [
+      zone == local.active_zones[0] ? {
+        name           = "subnet"
+        acl_name       = "hpc-acl"
+        cidr           = var.packer_subnet_cidr[0]
+        public_gateway = true
+        no_addr_prefix = local.no_addr_prefix
+      } : null
+    ] : []
+  }
+  subnets = { for zone, subnets in local.active_subnets : zone => [for each in subnets : each if each != null] }
+
+  # Use public gateway calculation
+  use_public_gateways = {
+    for zone in local.zones : zone => contains(local.active_zones, zone) ? true : false
+  }
+  network_acl_inbound_rules = [
+    {
+      name        = "test-1"
```

Comment:
These will be real rules on customer accounts, right? You probably shouldn't use the name test

Reply:
we can rename it

Reply:
@daniel-butler-irl As discussed we shall change the name here